### DISCUSSION TITLE
Add doc in sub-table via action

### DISCUSSION THREAD

**Discussion by webwurm** [AntlerVC/firetable#936] [Support / Q&A] [answered]
I can't figure out how to add a document in a sub-table of the current row in an action-script.
This is what I came up with to add an entry to the sub-table "logHistory" in the current row:

`const action:Action = async ({row,ref,db,storage,auth,actionParams,user}) => {

  const newEntry = {
    finishedTask: "Test!",
    timestamp: Date.now,
  };

  ref.collection('logHistory').add(newEntry);
}`

Sadly I get the error:
`Cannot read properties of undefined (reading 'success')`

Can somebody point me in the right direction please. Thanks!

**Comment by shamsmosowi** [comment, marked as answer]
hi @webwurm , actionScripts has an expectation of a returned object 
you mainly need to add something like this to the end of your function
```
return {
success:true,
message:"entry created"
}
```
more detailed examples [here](https://docs.rowy.io/field-types/action#examples)

**Reply by webwurm** [reply]
Thanks a lot, that works perfectly:

```
const action:Action = async ({row,ref,db,storage,auth,actionParams,user}) => {

  const newEntry = {
    finishedTask: "Test!",
    timestamp: new Date(),
  };

  ref.collection('logHistory').add(newEntry);

  return {
    success: true,
    message: `Log-Eintrag ${newEntry.finishedTask} erstellt`
  }

}
```
